runtime/ops/normalization.cc

```diff
@@ -127,7 +127,20 @@ std::tuple<chainerx::Array, XCVMOpaque*, chainerx::Array, chainerx::Array, chain
             x.device().GetBatchNormForwardBackward(result.mean, result.var, epsilon, decay, result.sorted_axis);
     const Array& gamma_reshaped = result.gamma;
     const Array& beta_reshaped = result.beta;
+
```

Comment:
I'm not 100% sure, but I think we can copy `mean` and `var` before passing them to `PreprocessBatchNorm`? I.e.,
```
PreprocessBatchNormResult result;
if (in_recomputing) {
  result = PreprocessBatchNorm(x, s, bias, mean.Copy(), var.Copy(), axes);
} else {
  result = PreprocessBatchNorm(x, s, bias, mean, var, axes);
}
```
In this way, we won't need to revert the change since the inplace modification will be done for copied mean/var?

Reply:
It sounds much better. I would try.

Reply:
updated.